File: backend_java/test-cdc/src/main/java/be/swsb/makeastats/testcdc/HttpBaseTest.java
package be.swsb.makeastats.testcdc;

import be.swsb.makeastats.MakeAStatsApplication;
import com.jayway.restassured.module.mockmvc.RestAssuredMockMvc;
import org.junit.Before;
import org.junit.runner.RunWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
import org.springframework.web.context.WebApplicationContext;

import javax.inject.Inject;

@RunWith(SpringJUnit4ClassRunner.class)
@SpringBootTest(
        classes = MakeAStatsApplication.class,
        webEnvironment = WebEnvironment.RANDOM_PORT)
public class HttpBaseTest {

    @Inject
    private WebApplicationContext webApplicationContext;

    @Before
    public void before() {
        RestAssuredMockMvc.webAppContextSetup(webApplicationContext);
    }
}

File: backend_kotlin/test-cdc/src/main/kotlin/be/swsb/makeastats/testcdc/HttpBaseTest.kt
package be.swsb.makeastats.testcdc

import be.swsb.makeastats.kotlinbackend.MakeAStatsApplication
import com.jayway.restassured.module.mockmvc.RestAssuredMockMvc
import org.junit.Before
import org.junit.runner.RunWith
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit4.SpringRunner
import org.springframework.web.context.WebApplicationContext
import javax.inject.Inject

@RunWith(SpringRunner::class)
@SpringBootTest(
        classes = arrayOf(MakeAStatsApplication::class),
        webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
open class HttpBaseTest {

    @Inject
    private val webApplicationContext: WebApplicationContext? = null

    @Before
    fun before() {
        RestAssuredMockMvc.webAppContextSetup(webApplicationContext)
    }
}